### PR TITLE
Change RichDate to value class (AnyVal)

### PR DESCRIPTION
**Pull request by oeddyo**
Change RichDate to value class (AnyVal)

https://github.com/twitter/scalding/issues/1120

### CODE REVIEW

File: scalding-date/src/test/scala/com/twitter/scalding/DateTest.scala [hunk 1]
```diff
@@ -66,11 +66,6 @@ class DateTest extends WordSpec {
       val rd2: RichDate = "  2010-10-02  T  00:00:01    "
       assert(rd1 === rd2)
     }
-    "Have same equals & hashCode as Date (crazy?)" in {
```
Comment (johnynek) [inline scalding-date/src/test/scala/com/twitter/scalding/DateTest.scala]: yes. This was crazy.

**Comment by johnynek** [comment]
this looks good to me.

**Comment by johnynek** [comment]
Related to #1120

**Comment by oeddyo** [comment]
Found an interesting build error:

https://travis-ci.org/twitter/scalding/jobs/56167536#L1

Maybe I somehow included a correct jar somehow? Trying to rebuild



error: error while loading <root>, zip file is empty
error: scala.reflect.internal.MissingRequirementError: object scala.runtime in compiler mirror not found.
	at scala.reflect.internal.MissingRequirementError$.signal(MissingRequirementError.scala:16)
	at scala.reflect.internal.MissingRequirementError$.notFound(MissingRequirementError.scala:17)
	at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:48)
	at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:40)
	at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:61)
	at scala.reflect.internal.Mirrors$RootsBase.getPackage(Mirrors.scala:172)
	at scala.reflect.internal.Mirrors$RootsBase.getRequiredPackage(Mirrors.scala:175)
	at scala.reflect.internal.Definitions$DefinitionsClass.RuntimePackage$lzycompute(Definitions.scala:183)
	at scala.reflect.internal.Definitions$DefinitionsClass.RuntimePackage(Definitions.scala:183)
	at scala.reflect.internal.Definitions$DefinitionsClass.RuntimePackageClass$lzycompute(Definitions.scala:184)
	at scala.reflect.internal.Definitions$DefinitionsClass.RuntimePackageClass(Definitions.scala:184)
	at scala.reflect.internal.Definitions$DefinitionsClass.AnnotationDefaultAttr$lzycompute(Definitions.scala:1024)
	at scala.reflect.internal.Definitions$DefinitionsClass.AnnotationDefaultAttr(Definitions.scala:1023)
	at scala.reflect.internal.Definitions$DefinitionsClass.syntheticCoreClasses$lzycompute(Definitions.scala:1153)
	at scala.reflect.internal.Definitions$DefinitionsClass.syntheticCoreClasses(Definitions.scala:1152)
	at scala.reflect.internal.Definitions$DefinitionsClass.symbolsNotPresentInBytecode$lzycompute(Definitions.scala:1196)
	at scala.reflect.internal.Definitions$DefinitionsClass.symbolsNotPresentInBytecode(Definitions.scala:1196)
	at scala.reflect.internal.Definitions$DefinitionsClass.init(Definitions.scala:1261)
	at scala.tools.nsc.Global$Run.<init>(Global.scala:1290)
	at scala.tools.nsc.Driver.doCompile(Driver.scala:32)
	at scala.tools.nsc.Main$.doCompile(Main.scala:79)
	at scala.tools.nsc.Driver.process(Driver.scala:54)
	at scala.tools.nsc.Driver.main(Driver.scala:67)
	at scala.tools.nsc.Main.main(Main.scala)
[SUGGESTION]: Try scald.rb --clean, you may have corrupt jars lying around
The command "scripts/test_tutorials.sh" exited with 127.
Done. Your build exited with 1.

**Comment by oeddyo** [comment]
Shall we merge this? or shall I complete the same thing for Duration and then merge? @johnynek

**Comment by johnynek** [comment]
Looks good. I don't see what the build error is, but it looks green to me.

**Pull request merged**